### PR TITLE
IpythonMagic: Rewrite deprecated imp.load_dynamic using importlib

### LINKED ISSUE #4640: [BUG]  IpythonMagic.py uses deprecated `imp` module
Should be replaced with importlib with fallback to `imp` for python version < 3.5

Note: Module uses function `imp.load_dynamic()` function. The same function is used in Inline.py file and is already ported to importlib machinery. Maybe it is enough just to move the `load_dynamic()` code to another module and use it in both modules.

### PR DESCRIPTION
**Pull request by matusvalo**
This PR rewrites deprecated `imp.load_dynamic()` with importlib-based implementation. It is based on example https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly. The logic of example is changed to accept only Extension files to reflect behaviour of `imp.load_dynamic()`.

Fixes #4640

### CODE REVIEW

**Comment by matusvalo** [comment]
Hmm it seems that the simmilar implementation is present here:
https://github.com/cython/cython/blob/6cd64875336b5121e145a3fe85c7621f3db5a64e/Cython/Build/Inline.py#L44-L50

So I think it is good idea to move the code somewhere to be shared. The implementation are the same except the implementation in Inline.py is:
* loading extensions and source file (mine is limited to extension only - it is same logic as `load_dynamic()`
* the module is not stored in `sys.modules` (mine is putting imported module - it is same logic as `load_dynamic()`)

**Comment by da-woods** [comment]
> Where we can put shared implementation? I was thinking in Cython.Utils. What do you think?

I think it should stay in Cython.Build. Create a file if you want and there isn't a suitable one. Maybe Cython.Build.Shared?

**Comment by matusvalo** [comment]
> Maybe Cython.Build.Shared?

I took a liberty and name it as `Utils.py`. Let me know if you are fine with the changes.

**Comment by scoder** [comment]
> So I think it is good idea to move the code somewhere to be shared. Where we can put shared implementation?

Note that `IpythonMagic.py` already imports `Inline.py`, which makes sense since both do very similar things. Just leave the shared code in `Inline.py` and import it from there. I'd specifically like to avoid adding a new module for basically nothing.

File: Cython/Build/Inline.py [hunk 1]
```diff
@@ -32,20 +32,19 @@ def to_unicode(s):
...
-    def load_dynamic(name, module_path):
-        return imp.load_dynamic(name, module_path)
-else:
-    import importlib.util as _importlib_util
-    def load_dynamic(name, module_path):
-        spec = _importlib_util.spec_from_file_location(name, module_path)
-        module = _importlib_util.module_from_spec(spec)
-        # sys.modules[name] = module
+    def load_dynamic(name, path):
+        spec = importlib.util.spec_from_file_location(name, loader=ExtensionFileLoader(name, path))
```
Comment (matusvalo) [inline Cython/Build/Inline.py:46]: I have adjusted the implementation in a way to limit importing only Extension modules.

**Comment by matusvalo** [comment]
I have implemented the comments. Could you please review again?

**Comment by scoder** [comment]
Looks good. I'll just wait for CI to finish before merging this.

**Comment by scoder** [comment]
Thanks

**Pull request merged**